remo/reports/models.py

```diff
@@ -320,23 +319,22 @@ def create_update_passive_event_report(sender, instance, created, **kwargs):
                                     instance.region,
                                     instance.country),
         'link': get_event_link(instance),
-        'activity_description': instance.description}
-
-    if created:
-        activity = Activity.objects.get(name=ACTIVITY_EVENT_CREATE)
-        attrs.update({
-            'user': instance.owner,
-            'event': instance,
-            'activity': activity,
-            'is_passive': True})
-
-        report = NGReport.objects.create(**attrs)
-        report.functional_areas.add(*instance.categories.all())
+        'activity_description': instance.description,
+        'campaign': instance.campaign,
+        'activity': activity,
+        'user': instance.owner,
+        'event': instance,
+        'is_passive': True,
+        'mentor': instance.owner.userprofile.mentor
+    }
+
+    report, report_created = NGReport.objects.get_or_create(**attrs)
+    if report_created:
+        (NGReport.objects.filter(event=instance, activity=activity)
+                         .exclude(pk=report.id).delete())
         statsd.incr('reports.create_passive_event')
     else:
```

Comment:
It doesn't really makes sense to update in this else ``case`` since you are using ``get_or_create``.